### DISCUSSION TITLE
image_processing/vips requires libvips 8.6+

### DISCUSSION THREAD

**Discussion by weefunker** [solidusio/solidus#4689] [Troubleshooting] [answered]
Happens on every new Rails app I try to install Solidus on

latest solidus version  3.2

1. bundle add solidus
2. rails g solidus:install
3. yes yes, defaults for frontend and admin
```
Loading images for Solidus Hoodie Zip
vips warning: VipsJpeg: error reading resolution
vips warning: VipsJpeg: error reading resolution
Loading images for Ruby Hoodie
Loading images for Ruby Hoodie Zip
vips warning: VipsJpeg: error reading resolution
vips warning: VipsJpeg: error reading resolution
Loading images for Ruby Polo
vips warning: VipsJpeg: error reading resolution
vips warning: VipsJpeg: error reading resolution
Loading images for Solidus Mug
Loading images for Ruby Mug
vips warning: VipsJpeg: error reading resolution
vips warning: VipsJpeg: error reading resolution
Loading images for Solidus Tote
Loading images for Ruby Tote
vips warning: VipsJpeg: error reading resolution
vips warning: VipsJpeg: error reading resolution
Loaded Assets samples
vips warning: VipsJpeg: error reading resolution
vips warning: VipsJpeg: error reading resolution
vips warning: VipsJpeg: error reading resolution
vips warning: VipsJpeg: error reading resolution
Loaded Addresses samples
Loaded Stores samples
Loaded Orders samples
Loaded Payments samples
Loaded Reimbursements samples


```

Vips warning on install after 
```
gem install ruby-vips
sudo apt install libvips libvips-dev
sudo apt install libvips-tools
```

None of these fixes the issue. Was only able to resolve this by setting the variant processor to imagemagick instead of libvips

#application.rb
config.active_storage.variant_processor = :mini_magick

running the server (produces an unspecified error)
running again app loads 
then removing the line from application.rb 
app again runs as normal until you try and view the product page and the error returns

Seems like something in the setup is interrupted by not being able to read the resolution of the starter template images 

Also curiously bin/dev will not work on any rails app with solidus installed 

As of writing the current version of lipvips is [v8.13.2](https://github.com/libvips/libvips/releases/tag/v8.13.2)

Unsure why or how it is asking for a future 8.6+ version

**Comment by gsmendoza** [comment]
Hello @weefunker! I wasn't able to reproduce the issue you're having. Please see the test app I created: https://github.com/gsmendoza/solidus_store/tree/rails-7-0-4--solidus-3-2-2--4686.

Can you provide the following info? It would help in debugging this issue.

1. What is the exact error message you're getting?
2. When are you seeing the error message? Based on the issue description, I can't tell if it's during install or when you run the server.
3. What is your version of  libvips? When I run `sudo apt install libvips libvips-dev`, I'm able to see that mine is 8.12.1:

```
solidus_store 18:08:22 $ sudo apt install libvips libvips-dev
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
Note, selecting 'libvips42' instead of 'libvips'
libvips-dev is already the newest version (8.12.1-1build1).
libvips42 is already the newest version (8.12.1-1build1).
libvips42 set to manually installed.
```

4. What is your OS version?

Thanks!

**Comment by weefunker** [comment, marked as answer]
Seems like the only solution was to upgrade my Ubuntu install from 18.04 to 22.04, lipvips version would not install on 18.04 without manually building

**Reply by gsmendoza** [reply]
Good to hear you figured out the issue! :)
